### DISCUSSION TITLE
Is there a way to identify and extract amalgamation of bars, lines and axis?

### DISCUSSION THREAD

**Discussion by DailyJK** [pymupdf/PyMuPDF#984] [Q&A]
I am trying to extract graphs from pdf. It’s easy to extract graphs that are image objects. Yet, some charts are not in image formats ([example](https://ibb.co/MpF2W4R)). One possible way I can think of is to identify an area where a chart exists and crop the area into an image. Is there any utility implementing a similar functionality? Thanks.

**Comment by JorjMcKie** [comment]
I suppose you mean the line chart at the example's top?
There is method `page.get_drawings()`, which extracts elementary draw commands and returns a list of such objects: lines, curves, rectangles.
You have a choice to either reconstruct them on some other PDF page using the methods of PyMuPDF's `Shape` class, or use their coordinates to create images of parts of the full PDF page using the `clip` parameter.
